src/utils/run_reports.py

```diff
@@ -0,0 +1,71 @@
+import logging
+import time
+
+
+def publish_report(wrapper, name, version):
+    try:
+        logging.info("Publishing report version {}-{}".format(name, version))
+        return wrapper.execute(wrapper.ow.publish_report, name, version)
+    except Exception as ex:
+        logging.exception(ex)
+        return False
+
+
+def run_reports(wrapper, config, reports, success_callback):
```

Comment:
The only change to this function is that instead of hard-coding the sending of the diagnostic report email on success, we pass an arbitrary success callback function that is called if a report runs successfully: https://github.com/vimc/montagu-task-queue/pull/12/files#diff-64022c19694443d8a984c9bbe62b6cbeR53

This means that this function can easily be re-used by any future tasks that involve running reports where the success callback might be a post to a slack hook, a different email being sent, some other task being triggered etc.

Reply:
This is neat - I might steal this pattern for doing orderly notifications on success.